cloudinit/distros/netbsd.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (C) 2019-2020 Gonéri Le Bouder
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import crypt
+import os
+import platform
+import six
+
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import ssh_util
+from cloudinit import util
+from cloudinit.distros import bsd_util
+
+LOG = logging.getLogger(__name__)
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/rc.conf'
+    ci_sudoers_fn = '/usr/pkg/etc/sudoers.d/90-cloud-init-users'
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'netbsd'
+        cfg['ssh_svcname'] = 'sshd'
+
+    def _read_system_hostname(self):
```

Comment:
This should probably be the same impelmentation as distro/debian
```suggestion
    def _read_system_hostname(self):
        sys_hostname = self._read_hostname(self.hostname_conf_fn)
        return (self.hostname_conf_fn, sys_hostname)
```

Reply:
It's rather different, the hostname is in /etc/rc.conf (RC file) and it's a key named `hostname`.